crates/libs/sys/src/core/mod.rs

```diff
@@ -46,5 +49,26 @@ macro_rules! link {
     )
 }
 
+#[cfg(windows_delay_load)]
+#[macro_export]
+macro_rules! link {
+    ($library:literal $abi:literal $(#[$($doc:tt)*])* fn $name:ident($($arg:ident: $argty:ty),*)->$ret:ty) => (
+        #[cfg(target_arch = "x86")]
+        type $name = ::core::option::Option<unsafe extern $abi fn($($argty),*) -> $ret>;
+        #[cfg(not(target_arch = "x86"))]
+        type $name = ::core::option::Option<unsafe extern "system" fn($($argty),*) -> $ret>;
+        pub unsafe fn $name($($arg: $argty),*) -> $ret {
+            static mut CACHE: $name = None;
```

Comment:
This is almost a nitpick but `static mut` is very difficult to use correctly (to the point where there is a strong possibility it will be deprecated and removed in a future edition). I think it would be better to use a `static AtomicPtr<{fn ptr type}>` instead. Acquire/Release are nearly free on x86 IIRC so there shouldn't be too much of a performance difference.